server/state_manager.js

```diff
@@ -0,0 +1,25 @@
+/*global JS_ENV Map require*/
+
+import { createStore } from 'redux';
+import { install } from 'redux-loop';
+import { fromJS } from 'immutable';
+
+export default class ServerStateManager {
+
+  initializeStore(initial_state,reducer){
+    this.store = createStore(reducer, initial_state, install());
+  }
+
+  // overridden in client state_manager.
+  initialState(opts, cookies){
+    return Object.assign({
+      session: fromJS({ token: cookies.token || null })
+    }, opts);
+  }
+
+  getInitialData(){
+    return Promise.resolve();
+  }
+
+}
```

Comment:
Again this is BaseState Manager.

We will also want to add a method to parse initial location (from server or client) into Redux state when initialized:

```js
import queryString from 'query-string';

// ...

  parseInitialLocation(initial_location, router){
    let route = router.findRouteByPath(initial_location.pathname),
        location_state = {...initial_location, route_name: route.route_name};

    location_state.query = initial_location.query ||
      queryString.parse(initial_location.search);
    location_state.params = route.setParams(location_state);
    return location_state;
  }

/// this is then used  in intializeRouterAndState:

  initializeRouterAndStore(i18n, initial_location, reducer, cookies) {
    let state_manager = this,
        router = new Router(i18n),
        initial_state = state_manager.initialState({
          location: this.parseInitialLocation(initial_location, router)
        }, cookies), 
    // ...
  }

```